src/meta/Base.ts

```diff
@@ -0,0 +1,23 @@
+import Map from '@dojo/shim/Map';
+import { WidgetMeta, WidgetMetaProperties } from '../interfaces';
+
+export class Base implements WidgetMeta {
+	protected nodes: Map<string, HTMLElement>;
+
+	constructor(properties: WidgetMetaProperties) {
+		this.nodes = properties.nodes;
+		this.invalidate = properties.invalidate;
+		this.requireNode = properties.requireNode;
+	}
+
+	public has(key: string): boolean {
+		this.requireNode(key);
+		return this.nodes.has(key);
+	}
+
+	protected invalidate(): void {}
+
+	protected requireNode(key: string): void {}
```

Comment:
Subclasses will call this directly to notify the widget system they require that node.